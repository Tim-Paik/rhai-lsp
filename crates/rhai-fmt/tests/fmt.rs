use pretty_assertions::assert_eq;

macro_rules! assert_fmt {
    ($src:expr) => {
        {
            let s = $src;
            assert_eq!(s, rhai_fmt::format_source(s, Default::default()));
        }
    };
    ($src:expr, $expected:expr) => {
        {
            let s = $src;
            assert_eq!($expected, rhai_fmt::format_source(s));
        }
    };
}

#[test]
fn fmt_smoke() {
    let _ = tracing_subscriber::fmt().with_max_level(tracing::Level::DEBUG).try_init();

    assert_fmt!("/// hello
/// there
let a = 1234;
let b = 'a'
");
}
