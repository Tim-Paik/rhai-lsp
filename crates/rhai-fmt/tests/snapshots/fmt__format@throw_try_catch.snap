---
source: crates/rhai-fmt/tests/fmt.rs
expression: formatted
---
try {
  throw "hello";
} catch {}

